compiler/rustc_resolve/src/macros.rs

```diff
@@ -777,7 +782,8 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
 
             if trace {
                 let kind = kind.expect("macro kind must be specified if tracing is enabled");
-                self.multi_segment_macro_resolutions.push((
+                // FIXME: Should be an output of Speculative Resolution.
```

Comment:
As I said, I'm not sure if it needs to be an output.
It can be if it's easy to do, but it would be enough if it just was under a mutex.